R/fetch_table.R

```diff
@@ -0,0 +1,62 @@
+#' Fetch Exoplanets table
+#'
+#' @details
+#' Fetches data from an exoplanets TAP (Table Access Protocol) service
+#' and returns it as a data frame.
+#' You can optionally specify `WHERE` ADQL clause to filter rows based on conditions.
+#'
+#' @param table A string specifying the table to query. Must be one of:
+#'   \itemize{
+#'   \item `ps` - Planetary Systems table
+#'   \item `pscomppars` – Planetary Systems Composite Parameters table
+#'   \item `stellarhosts` – Stellar Hosts table
+#'   \item `keplernames` – Kepler Confirmed Names
+#' }
+#' @param filters Optional ADQL WHERE clause as a string, e.g., "pl_bmasse > 1 AND st_teff < 6000".
+#'
+#' @returns A data frame containing fetched data.
+#'
+#' @importFrom httr2 request req_perform resp_body_string req_options
+#' @importFrom readr read_csv
+#' @importFrom dplyr `%>%`
+#' @importFrom checkmate assert_choice assert_string
+#'
+#' @examples
+#' \dontrun{
+#'  # All entries from Stellar Hosts table
+#'  fetch_table("stellarhosts")
+#'  # Entries from Planetary Systems table where planetary mass > 3 times the earth mass
+#'  fetch_table("ps", filters = "pl_bmasse > 3")
+#'  # Planets orbiting Teegarden's Star with radius > 1 Earth radius
+#'  fetch_table("pscomppars", filters = "hostname = 'Teegarden''s Star' and pl_rade > 1")
+#' }
+#'
+#' @export
+fetch_table = function(table, filters = NULL) {
```

Comment:
issue: This argument does not really qualify as `filters`, should be `query` or `query_string`. This is literally a query string that is passed.
```suggestion
fetch_table = function(table, query_string = NULL) {
```